src/NKSolver/NKSolvers.F90

```diff
@@ -3075,25 +3094,41 @@ subroutine physicalityCheckANK(lambdaP)
              do k=2, kl
                 do j=2, jl
                    do i=2, il
-                      ! multiply the ratios by 10 to check if the change in a
-                      ! variable is greater than 10% of the variable itself.
+                      ! multiply the ratios by ANK_physLSTol to check if the change in a
+                      ! variable is greater than ANK_physLSTol of the variable itself.
 
                       ! check density
+#ifndef USE_COMPLEX
+                      ! to have the real mode sliiiightly more efficient, just do stuff with real numbers
                       ratio = abs(wvec_pointer(ii)/(dvec_pointer(ii)+eps))*ANK_physLSTol
+#else
+                      ! We dont care what happens to the complex part of the update because
+                      ! that is a linear system. So again check the real update for the physical
+                      ! line search. Towards the end of the simulation, real part gets smaller and
+                      ! and smaller updates, so this routine will always give a step of 1 which is what
+                      ! we want for the complex parts.
+                      ratio = abs(real(wvec_pointer(ii)) / real(dvec_pointer(ii) + eps)) * real(ANK_physLSTol)
+#endif
                       lambdaL = min(lambdaL, ratio)
 
                       ! increment by 4 because we want to skip momentum variables
                       ii = ii + 4
 
                       ! check energy
+#ifndef USE_COMPLEX
+                      ! see the comment above for the difference between real and complex versions
                       ratio = abs(wvec_pointer(ii)/(dvec_pointer(ii)+eps))*ANK_physLSTol
+#else
+                      ratio = abs(real(wvec_pointer(ii))/real(dvec_pointer(ii)+eps))* real(ANK_physLSTol)
```

Comment:
So I was just trying to failsafe things. I realized that I could not set fortran-level options from pyADflow in the complex mode using the `alwaysRealType` for the data type. As a result, even though I would have preferred to have these variables be always real, I have to use `realType` for them. Locally, the complex part is set to zero by default, but I am not sure if all compilers would give this result so I just check. Do you have a better suggestion here @joanibal?

Reply:
>  I realized that I could not set fortran-level options from pyADflow in the complex mode using the alwaysRealType for the data type

I have done this. I could be misunderstanding what you mean though.

Reply:
I'd feel a lot better about the PR if the step lengths were always real

Reply:
Can you show me how you did it? Specifically, I want to have a pyADflow option that is a float, to be passed to an `alwaysRealType` variable in Fortran. I think when I changed the variable type to `alwaysRealType` both in the actual code and the pyadflow.pyf file, it _runs_ but it was passing random noise between the variables. The target variables in fortran ended up getting zeros most of the time if I remember correctly. Let me know if you know the solution to this issue and I can fix it. We can also make it a separate PR because there are actually a few variables that are supposed to be always real but we just use the real type because we set them from python.

Reply:
Okay so @joanibal implemented this `alwaysRealType` passing with f2py in his development branch. I created an issue detailing the fix here: https://github.com/mdolab/adflow/issues/133

@joanibal, I think we should do a separate PR for this since that possibly touches more than just these solvers. I think the solvers do work as is, so I think this PR should be fine. I agree with you though, it would be better if we could use `alwaysRealType` for these variables.